.github/workflows/free-threaded-tests.yml

```
@@ -0,0 +1,261 @@
+---
+    name: Free-threaded tests
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.event.number }}-${{ github.event.ref }}
+      cancel-in-progress: true
+
+    on:
+      push:
+        branches-ignore:
+          - auto-backport-of-pr-[0-9]+
+          - v[0-9]+.[0-9]+.[0-9x]+-doc
+          - dependabot/**
+      pull_request:
+        branches-ignore:
+          - v[0-9]+.[0-9]+.[0-9x]+-doc
+        paths-ignore:
+          # Skip running tests if changes are only in documentation directories
+          - 'doc/**'
+          - 'galleries/**'
+      schedule:
+        # 5:47 UTC on Saturdays
+        - cron: "47 5 * * 6"
+      workflow_dispatch:
+        workflow: "*"
+
+    env:
+      NO_AT_BRIDGE: 1  # Necessary for GTK3 interactive test.
+      OPENBLAS_NUM_THREADS: 1
+      PYTHONFAULTHANDLER: 1
+
+    jobs:
+      test:
+        if: >-
+          github.event_name == 'workflow_dispatch' ||
+          (
+            github.repository == 'matplotlib/matplotlib' &&
+            !contains(github.event.head_commit.message, '[ci skip]') &&
+            !contains(github.event.head_commit.message, '[skip ci]') &&
+            !contains(github.event.head_commit.message, '[skip github]') &&
+            !contains(github.event.head_commit.message, '[ci doc]')
+          )
+        permissions:
+          contents: read
+        name: "Python 3.13t on ubuntu-latest (free-threaded)"
+        runs-on: ubuntu-latest
+
+        steps:
+          - uses: actions/checkout@v4
+            with:
+              fetch-depth: 0
+
+          - name: Set up Python 3.13t
+            uses: deadsnakes/action@6c8b9b82fe0b4344f4b98f2775fcc395df45e494 # v3.1.0
+            with:
+                python-version: "3.13"
+                nogil: true
```

```suggestion
            uses: deadsnakes/action@6c8b9b82fe0b4344f4b98f2775fcc395df45e494  # v3.1.0
            with:
              python-version: "3.13"
              nogil: true
```

In addition, the whole file should be dedented by 4 spaces (aside from the first line which has no spaces)